libvdrtools/indy-wallet/src/lib.rs

```diff
@@ -704,6 +703,65 @@ impl WalletService {
         Ok(())
     }
 
+    pub async fn migrate_records<E>(
+        &self,
+        old_wh: WalletHandle,
+        new_wh: WalletHandle,
+        mut migrate_fn: impl FnMut(Record) -> Result<Option<Record>, E>,
+    ) -> IndyResult<()>
+    where
+        E: std::fmt::Display,
+    {
+        let old_wallet = self.get_wallet(old_wh).await?;
+        let new_wallet = self.get_wallet(new_wh).await?;
+
+        let mut records = old_wallet.get_all().await?;
+        let mut num_records = 0;
+
+        while let Some(WalletRecord {
+            type_,
+            id,
```

Comment:
Given that we found that the wallet schema is expected to have`id` field, I would suggest to rename this field in `WalletRecord` from `id` to `name`. Because the `id` field in WalletRecord / StorageRecord structs is originating from `name` column in the db schema.

<img width="770" alt="image" src="https://github.com/hyperledger/aries-vcx/assets/2149283/0e8085be-8d26-434b-a888-5cd266159bdd">

Reply:
I'd do the opposite. I'd rename the name column to id, because that's exactly what gets passed in.

Reply:
ahh I see you've removed the `id` autoincremented field, so it's possible. Okay then - although being renamed to `id` I would usually expect that to be the primary key, which it won't be - I'll leave the final decision to your judgement.

Reply:
Well, I looked further into this and the problem is that if we were to update the `name` column to `id`, we should do that everywhere, including in the `SQLite` impl.

But the `SQLite` impl has an `id` field already, and it's being used for other purposes too. So... maybe we should just leave it like this. We're not really aiming to maintain libvdrtools, right? So someday I guess we'll want to drop even the `Indy` wallet.

And I'm also saying this because this is the kind of "problem" that wouldn't have surfaced if not for the modifications I made to `libvdrtools` to assist with the migration.

Reply:
okay, makes sense. Since we removed the `id` autoincrement in the schema, that mostly eliminates the potential confusion I was pointing out in my original comment.